### PR TITLE
New packages: cross-x86_64-w64-mingw32 and cross-i686-w64-mingw32

### PR DESCRIPTION
**Pull request by alekseyt**
Two cross-compilers:

* x86_64-w64-mingw32
* i686-w64-mingw32

Template builds both.

Based on GCC 7.3.0 and binutils 2.29.1 (synced with current's GCC). Other components versions: gmp, mpfr, mpc, isl are also synced with `gcc/template`. MinGW-w64 is latest (5.0.4).

Exceptions support - yes (sjlj in i686 and seh in x86_64), threads support - yes (winpthreads). I've checked that compilers work on more or less sophisticated application (SDL, ogg, flac, OpenGL, glm, etc).

License (ZPL) is included into packages.

There also were similar package requests in old issue tracker: 

* https://github.com/voidlinux/void-packages/issues/390
* https://github.com/voidlinux/void-packages/issues/3895

### CODE REVIEW

**Comment by abenson** [comment]
Initial testing went well. Tomorrow I'll test one of my larger projects!

File: srcpkgs/cross-x86_64-w64-mingw32/template [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+_isl_version=0.16.1
+_mingw_version=5.0.4
+
+pkgname=cross-x86_64-w64-mingw32
+version=${_mingw_version}
+revision=1
+short_desc="Cross toolchain for Win64"
+maintainer="Aleksey Tulinov <aleksey.tulinov@gmail.com>"
+homepage="https://sourceforge.net/projects/mingw-w64/"
+license="LGPL-3, GPL-3, ZPL"
```
Comment (abenson) [inline srcpkgs/cross-x86_64-w64-mingw32/template]: SPDX identifiers: `GPL-3.0-only, LGPL-3.0-only, ZPL-2.1`

File: srcpkgs/cross-x86_64-w64-mingw32/template [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+_gmp_version=6.1.2
+_mpfr_version=4.0.0
+_mpc_version=1.1.0
+_isl_version=0.16.1
+_mingw_version=5.0.4
+
+pkgname=cross-x86_64-w64-mingw32
+version=${_mingw_version}
+revision=1
+short_desc="Cross toolchain for Win64"
```
Comment (abenson) [inline srcpkgs/cross-x86_64-w64-mingw32/template]: Nitpicking, but Win32/64

**Comment by abenson** [comment]
There seems to be a few file conflicts with gcc introduced. 

~~~
 /usr/lib/libcc1.so.0.0.0
 /usr/share/info/cpp.info.gz
 /usr/share/info/cppinternals.info.gz
 /usr/share/info/gcc.info.gz
 /usr/share/info/gccinstall.info.gz
 /usr/share/info/gccint.info.gz
 /usr/share/man/man7/fsf-funding.7
 /usr/share/man/man7/gfdl.7
 /usr/share/man/man7/gpl.7
 /usr/lib/libcc1.so -> /usr/lib/libcc1.so.0.0.0
 /usr/lib/libcc1.so.0 -> /usr/lib/libcc1.so.0.0.0
~~~

**Comment by abenson** [comment]
Looks like in `_install_cross()` you can just add:

~~~
   # move libcc1.so* to the sysroot
   mv ${DESTDIR}/usr/lib/libcc1.* ${DESTDIR}/${_sysroot}/usr/lib

   # Remove unnecessary stuff
   rm -rf ${DESTDIR}/usr/share
~~~

That should cover everything I've seen so far

**Comment by alekseyt** [comment]
Thank you for looking into this. Actually there are some more changes pending with gmp, mpfr, mpc and isl from current (like in other cross-compilers) instead of downloading and building them separately, also some template formatting changes to better fit established practices (i think). But i had to check those changes first, which i did, but didn't update PR yet.

Sorry about conflicting files, i thought i checked the packages but apparently i missed this issue. If there are other issues with template that i overlooked - i'll try to cover all the comments in PR update(s).

**Comment by abenson** [comment]
Okay, cool. I look forward to it, it's building my bigger project (although quite small still) perfectly. Running great. Can't wait!

**Comment by alekseyt** [comment]
On second thoughts, maybe it's better to have gmp, mpfr, mpc and isl built specifically for these cross-compilers. This way their update in current won't break the package. I still can sync GCC version with current if needed, MinGW builds with GCC 8.2 and current's `-devel` packages, it's all fine, i don't expect any major problems on that front, but i do expect problems if, for instance, isl is updated unexpectedly.

Assuming this is used for work, availability and stability shall take priority IMO. I think it's better to leave toolchain components where they are for now. Otherwise this might create two more problems for two maintainers.

File: srcpkgs/cross-x86_64-w64-mingw32/template (hunk 1 above)
Comment (alekseyt) [reply srcpkgs/cross-x86_64-w64-mingw32/template]: This is fixed in e94ff0ec, but i believe it's `LGPL-3.0-or-later`. I've aligned licenses similarly to how they are specified in other cross-compilers.

**Comment by alekseyt** [comment]
Other changes:

* GCC version in packages description. MinGW version kind of shadows GCC version, this way a person looking at packages list don't have to guess or research. Example:

```
ii cross-i686-w64-mingw32-5.0.4_2                  Cross toolchain for Win32 (GCC 7.3.0)
ii cross-x86_64-w64-mingw32-5.0.4_2                Cross toolchain for Win64 (GCC 7.3.0)
```

* `only_for_archs="i686 x86_64"` - i don't know how this will play with musl so i thought it's better to limit list of platforms and extend it if needed rather than vice versa.
* Removed unnecessary stuff from packages (was causing file conflicts). I don't know why `libcc1` is moved from `/usr/lib` to sysroot in other cross-compilers, seems to be unused in MinGW, so now it's gone.
* Using `{GNU_SITE}` etc in URLs

**Comment by alekseyt** [comment]
Some more changes:

* GCC 8.2

This is just future-proofing of this PR. Other components are synced to what is coming for current's update. It might desync at some point, but i believe MinGW cross-compilers will build regardless.

* disabled `libcc1` in build *and* i also left `libcc1` cleanup in template, just in case if it will slip into packages somehow.
* sanitized toolchain build environment by using subshells where appropriate.

**Comment by Gottox** [comment]
Let's get this merged.

**Pull request merged**